drivers/iio/proximity/vcnl3020.c

```diff
@@ -0,0 +1,338 @@
+// SPDX-License-Identifier: GPL-2.0-only
+/*
+ * vcnl3020.c - Support for Vishay VCNL3020 proximity sensor
+ *
+ * based on vcnl4000.c
+ */
+
+#include <linux/module.h>
+#include <linux/i2c.h>
+#include <linux/err.h>
+#include <linux/delay.h>
+
+#include <linux/iio/iio.h>
+#include <linux/iio/sysfs.h>
+#include <linux/platform_device.h>
+#include <linux/iio/proximity/vcnl3020.h>
+
+static const struct i2c_device_id vcnl3020_id[] = {
+	{ "vcnl3020", 0 },
+	{}
+};
+MODULE_DEVICE_TABLE(i2c, vcnl3020_id);
+
+static int32_t vcnl3020_init(struct vcnl3020_data *data)
+{
+	s32 rc;
+	u32 proximity_rate, led_current, threshold, count_exceed, param_val;
+
+	struct device *dev = &data->client->dev;
+
+	rc = device_property_read_u32(dev, "proximity-rate", &param_val);
+	if (rc) {
+		dev_err(dev, "couldn't get proximity rate %x", rc);
```

Comment:
`"Couldn't`. Capitalize.

Reply:
yea, I'll change messages in next version.